src/main/java/org/opentripplanner/profile/RoundBasedProfileRouter.java

```diff
@@ -0,0 +1,487 @@
+package org.opentripplanner.profile;
+
+import gnu.trove.iterator.TObjectIntIterator;
+import gnu.trove.map.TObjectIntMap;
+import gnu.trove.map.hash.TObjectIntHashMap;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import jersey.repackaged.com.google.common.collect.Maps;
+import jersey.repackaged.com.google.common.collect.Sets;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Collections2;
+
+import org.geotools.xml.xsi.XSISimpleTypes.Int;
+import org.mapdb.Fun.Tuple2;
+import org.onebusaway.gtfs.model.Stop;
+import org.opentripplanner.analyst.TimeSurface;
+import org.opentripplanner.analyst.TimeSurface.RangeSet;
+import org.opentripplanner.api.parameter.QualifiedModeSet;
+import org.opentripplanner.common.model.GenericLocation;
+import org.opentripplanner.profile.ProfileState.Type;
+import org.opentripplanner.routing.algorithm.AStar;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.routing.core.TraverseMode;
+import org.opentripplanner.routing.edgetype.SimpleTransfer;
+import org.opentripplanner.routing.edgetype.TripPattern;
+import org.opentripplanner.routing.error.VertexNotFoundException;
+import org.opentripplanner.routing.graph.Edge;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.routing.spt.DominanceFunction;
+import org.opentripplanner.routing.spt.ShortestPathTree;
+import org.opentripplanner.routing.trippattern.FrequencyEntry;
+import org.opentripplanner.routing.trippattern.TripTimes;
+import org.opentripplanner.routing.vertextype.TransitStop;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Multimap;
+
+/**
+ * Profile routing using a round-based approach, more or less like RAPTOR (http://research.microsoft.com/pubs/156567/raptor_alenex.pdf)
+ * @author mattwigway
+ *
+ */
+public class RoundBasedProfileRouter {
```

Comment:
This is unused now, except in other unused functions. I guess we have some cleanup to do after deciding which techniques to retain. Of course profile routing for modeify is a different matter than departure-time-range analyst, so we'll need to discuss this.